### PR TITLE
Add isort to pre-commit hooks, package resorting

### PR DESCRIPTION
**Pull request by charlesbluca**
Second pass at #4600, as there were a lot of merge conflicts piling up there. 

- [ ] Tests added / passed
- [ ] Passes `black distributed` / `flake8 distributed`

### CODE REVIEW

**Comment by charlesbluca** [comment]
Looks like some failures due to deprecations in pytest - does this have something to do with #4646?

**Comment by jrbourbeau** [comment]
Yep, I was trying to figure out what's causing the test suite to start failing (definitely unrelated to the changes here). It looks like there was an issue in a new release of the `setup-miniconda` GitHub action (xref https://github.com/conda-incubator/setup-miniconda/issues/160), though this should be fixed soon for us with https://github.com/dask/distributed/pull/4585

**Comment by charlesbluca** [comment]
Good to know! In that case, I'll revisit the tests once that is merged.

**Comment by jrbourbeau** [comment]
https://github.com/dask/distributed/pull/4585 is in, so you should be able to resolve the test failures here by merging `main` into your PR branch

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 repos:
+  -   repo: https://github.com/pycqa/isort
+      rev: 5.7.0
```
Comment (jrbourbeau) [inline .pre-commit-config.yaml:3]: It looks like release came out a few days ago

```suggestion
      rev: 5.8.0
```

File: distributed/__init__.py [hunk 2]
```diff
@@ -1,42 +1,42 @@
-from . import config
+from . import config  # isort:skip
```
Comment (jrbourbeau) [inline distributed/__init__.py:1]: Could you add a small comment here about why we need to skip `isort`?

**Review by jrbourbeau — COMMENTED** [review]
Thanks @charlesbluca!

I have a similar question to [this one](https://github.com/dask/dask/pull/7370#pullrequestreview-623569450) over in `dask/dask` about making `dask` a special third party library so it's imports are just above the `distributed` imports instead of being grouped with things like `numpy`, `tlz`, etc.

cc @jsignell

File: distributed/__init__.py (hunk 2 above)
Comment (charlesbluca) [reply distributed/__init__.py:1]: Sure!

**Comment by charlesbluca** [comment]
Thanks for the review! Distinguishing Dask from other imports makes sense to me, happy to add that in as long as its consistent with dask/dask#7370.

**Comment by jrbourbeau** [comment]
We just added a new module which is throwing some `isort` errors. Would you mind running `isort` on the new module and pushing up the changes?

**Comment by charlesbluca** [comment]
Sure! Also want to note that a test that failed in the original PR #4600, `test_as_completed_with_results`, seems to have worked in these subsequent CI runs, so maybe it should be marked as flaky.

**Comment by jsignell** [comment]
> Most of the test failures have something to do with human readable byte formatting not being consistent:

This has to do with the work that @crusaderky has been doing in https://github.com/dask/dask/pull/7484

**Review by jrbourbeau — APPROVED** [review]
Thanks @charlesbluca!

**Comment by charlesbluca** [comment]
Thanks for the help @jrbourbeau and @jsignell!

**Pull request merged**